### PR TITLE
!makes APIs discoverable and introduces scripted tests

### PR DESCRIPTION
**Pull request by ktoso**
WIP - polishing this one up a bit still.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,24 +69,24 @@ import NativePackagerKeys._
...
 _...otherwise you will get duplicate imports reported. This is because the new 1.0+ version uses sbt's auto plugin feature._
 
-Unless the ConductR is running at http://127.0.0.1:9005, and instead supposing that it is running at
-http://192.168.59.103:9005 then you will typically issue the following command from the sbt console:
+Unless the ConductR is running at `127.0.0.1:9005`, and instead supposing that it is running at
+`192.168.59.103:9005` then you will typically issue the following command from the sbt console:
 
 ```
-conductr http://192.168.59.103:9005
+conductr:set 192.168.59.103:9005
```
Comment (hseeberger) [inline README.md]: So `conductr` is a configuration now? Nice!
Yet I don't like `set`: Set what? Why not `control`?

Comment (ktoso) [inline README.md]: control sounds good :+1:

Comment (ktoso) [inline README.md]: or `control-server`

File: src/main/scala/com/typesafe/conductr/sbt/SbtTypesafeConductR.scala [hunk 2]
```diff
@@ -247,12 +263,12 @@ object SbtTypesafeConductR extends AutoPlugin {
...
   // Actor system management and API
 
-  private val actorSystemAttrKey = AttributeKey[ActorSystem]("sbt-typesafe-conductr-actor-system")
+  private val actorSystemAttrKey = AttributeKey[ActorSystem]("sbt-typesafe-setConductrTask-actor-system")
 
   private def loadActorSystem(state: State): State =
     state.get(actorSystemAttrKey).fold {
       state.log.debug(s"Creating actor system and storing it under key [${actorSystemAttrKey.label}]")
-      val system = withActorSystemClassloader(ActorSystem("sbt-typesafe-conductr"))
+      val system = withActorSystemClassloader(ActorSystem("sbt-typesafe-setConductrTask"))
```
Comment (hseeberger) [inline src/main/scala/com/typesafe/conductr/sbt/SbtTypesafeConductR.scala]: set-conductr-task, i.e. don't mix camel case and the other one (with hyphens).

File: version.sbt [hunk 3]
```diff
@@ -1 +1,12 @@
-version in ThisBuild := "0.26.0-SNAPSHOT"
\ No newline at end of file
+import java.io.FileInputStream
+import java.util.Properties
+
+version in ThisBuild := {
+  val s = new FileInputStream(file("version.properties"))
```
Comment (hseeberger) [inline version.sbt]: Nice trick!

File: README.md (hunk 1 above)
Comment (hseeberger) [inline README.md]: +1

File: src/main/scala/com/typesafe/conductr/sbt/SbtTypesafeConductR.scala (hunk 2 above)
Comment (ktoso) [inline src/main/scala/com/typesafe/conductr/sbt/SbtTypesafeConductR.scala]: right, thanks

**Comment by ktoso** [comment]
refs https://github.com/sbt/sbt-typesafe-conductr/issues/50

File: README.md [hunk 4]
```diff
@@ -28,7 +28,7 @@ A bundle will be produced from the native packager settings of this project. A b
 packager distribution and includes some component configuration. To load the bundle:
 
 ```bash
-loadBundle <HIT THE TAB KEY AND THEN RETURN>
+conductr:loadBundle <HIT THE TAB KEY AND THEN RETURN>
```
Comment (huntc) [inline README.md]: If we're going to change anything in terms of the structure of the commands then we should align more closely with the Python CLI - there's a ticket for this.

Thus the above command should be `conduct load`.

File: README.md [hunk 5]
```diff
@@ -69,24 +69,24 @@ import NativePackagerKeys._
...
 _...otherwise you will get duplicate imports reported. This is because the new 1.0+ version uses sbt's auto plugin feature._
 
-Unless the ConductR is running at http://127.0.0.1:9005, and instead supposing that it is running at
-http://192.168.59.103:9005 then you will typically issue the following command from the sbt console:
+Unless the ConductR is running at `127.0.0.1:9005`, and instead supposing that it is running at
+`192.168.59.103:9005` then you will typically issue the following command from the sbt console:
 
 ```
-conductr http://192.168.59.103:9005
+conductr:controlServer 192.168.59.103:9005
```
Comment (huntc) [inline README.md]: We can abolish this command as a result of now having `CONDUCTR_IP` and `CONDUCTR_PORT`.

**Comment by huntc** [comment]
@ktoso This PR has gone a little beyond what I had envisaged. I'm now hung up on the re-factoring of the commands. If we're going to do that then we need to align them with the Python CLI - I've received such feedback from others already. Can you please either remove the re-factoring of the commands or re-factor them such that they align with the Python CLI.

Thanks.

File: README.md (hunk 4 above)
Comment (ktoso) [inline README.md]: I argue for `conductr:load`, which it is currently. The above it a typo in docs.

File: README.md [hunk 6]
```diff
@@ -69,24 +69,24 @@ import NativePackagerKeys._
...
-Property     | Description
--------------|------------
-conductr     | Sets the ConductR's address to a provided url (defaults to env varialbe `CONDUCTR_IP:[CONDUCTR_PORT]` if set, otherwise uses the default: `http://127.0.0.1:9005`)
-loadBundle   | Loads a bundle and an optional configuration to the ConductR
-startBundle  | Starts a bundle given a bundle id with an optional absolute scale value
-stopBundle   | Stops all executions of a bundle given a bundle id
-unloadBundle | Unloads a bundle entirely (requires that the bundle has stopped executing everywhere)
+Property              | Description
+----------------------|------------
+conductr              | Sets the ConductR's address to a provided url (defaults to env varialbe `CONDUCTR_IP:[CONDUCTR_PORT]` if set, otherwise uses the default: `http://127.0.0.1:9005`)
```
Comment (ktoso) [inline README.md]: FIXME: this is controlServer

File: README.md (hunk 4 above)
Comment (huntc) [inline README.md]: If the goal is discoverability, then do you agree that a `conduct` command achieves this goal given tab completion?

I'm trying to understand why having a scope would be better than a command, given that the latter more closely resembles our Python CLI.

Comment (ktoso) [inline README.md]: You're right that it's doable with commands, was not the immediate "sbt style" I would think of. 
It's more work but can be done if you think it's worth it

File: README.md (hunk 5 above)
Comment (ktoso) [inline README.md]: related - CONDUCTR_*IP* seems wrong, isn't it "HOST" or "URL"? Can be a domain name, not just IP.

File: README.md (hunk 4 above)
Comment (huntc) [inline README.md]: I think that having the command align with the Python CLI as much as reasonable makes sense, and was raised by someone else. I realise that it is more work, but I think it is worth it. Putting the effort aside, let's ask ourselves what the best approach would be. WDYT?

Comment (ktoso) [inline README.md]: As a scala dev I'd expect `namespace:...` to include both tasks and other keys I can `inspect` on, not only the DSL as seen in command line.

CLI-like commands would be an addition, these could use `conduct`, while keys are all in `conductr`.

Comment (huntc) [inline README.md]: With sbt-web we chose the approach that common commands were not namespaced in order to reduce typing.

When I think about it, the `conduct` command for the Python CLI is really about namespacing on the command line, so an sbt convention of namespacing all commands to align with the CLI sub commands sounds reasonable.

Let's just fix up the doc then and go with the namespaced approach then. :-)

Comment (huntc) [inline README.md]: @ktoso Any chance we can get this PR finished off? Thanks.

Comment (ktoso) [inline README.md]: yes sure, I do mean to pull through my outstanding things of course :)
on this one just the typo in docs (where this comments thread is attached to) as I understand the current consensus?

File: README.md (hunk 5 above)
Comment (huntc) [inline README.md]: It must be an ip in other contexts eg for ConductR itself.

**Comment by huntc** [comment]
Looking good with exception to your fixme

File: README.md [hunk 7]
```diff
@@ -69,24 +69,23 @@ import NativePackagerKeys._
...
--------------|------------
-conductr     | Sets the ConductR's address to a provided url (defaults to env varialbe `CONDUCTR_IP:[CONDUCTR_PORT]` if set, otherwise uses the default: `http://127.0.0.1:9005`)
-loadBundle   | Loads a bundle and an optional configuration to the ConductR
-startBundle  | Starts a bundle given a bundle id with an optional absolute scale value
-stopBundle   | Stops all executions of a bundle given a bundle id
-unloadBundle | Unloads a bundle entirely (requires that the bundle has stopped executing everywhere)
+Property               | Description
+-----------------------|------------
+conductr:controlServer | Sets the ConductR's address to a provided url (defaults to env varialbe `CONDUCTR_IP:[CONDUCTR_PORT]` if set, otherwise uses the default: `http://127.0.0.1:9005`)
+conductr:loadBundle    | Loads a bundle and an optional configuration to the ConductR
```
Comment (huntc) [inline README.md]: @ktoso This looks as though it still needs changing. Any ETA? Thanks.

Comment (ktoso) [inline README.md]: ETA - now.
Rebased it as well.

**Pull request merged**